### DISCUSSION TITLE
"Token passed to createTokenAuth is not a string" in response to basic authentication attempt

### DISCUSSION THREAD

**Discussion by anthonyhaffey** [octokit/octokit.js#1881] [Q&A] [answered]
I'm new to Octokit, so I assume this is something very simple I'm misunderstanding/overlooking:

I'm trying to get a minimal example of authentication working in node js. However, when I run:

```
const { Octokit } = require('@octokit/rest');

const {
  createBasicAuth,
  createAppAuth,
  createOAuthAppAuth,
  createTokenAuth,
  createActionAuth,
} = require("@octokit/auth");



const octokit = new Octokit({
  auth: {
    username: this_username,
    password: this_password
  }
});
```

Or a lot of variations of it, I always get:

```
Uncaught Error: [@octokit/auth-token] Token passed to createTokenAuth is not a string
    at Object.createTokenAuth (D:\github\sos\collector-dev\web\node_modules\@octokit\auth-token\dist-node\index.js:39:11)
    at new Octokit (D:\github\sos\collector-dev\web\node_modules\@octokit\core\dist-node\index.js:115:32)
    at new _a (D:\github\sos\collector-dev\web\node_modules\@octokit\core\dist-node\index.js:166:30)
    at new OctokitWithDefaults (D:\github\sos\collector-dev\web\node_modules\@octokit\core\dist-node\index.js:141:9)'
```

I get this whether or not **this_password** is a valid password associated with **this_username**

The username and password I've tried are my github username and password. What am I missing?
Many thanks in advance

**Comment by anthonyhaffey** [comment]
Actually - it looks like using username and password is going to become deprecated anyway - I'll look into making the authorization token way work

**Comment by gr2m** [comment, marked as answer]
You also have to pass `createBasicAuth` to the `Octokit` constructor

```js
const octokit = new Octokit({
  authStrategy: createBasicAuth,
  auth: {
    username: this_username,
    password: this_password
  }
});
```

But as you mentioned, basic auth is deprecated for the API and will soon be removed, I recommend against relying on it

**Reply by oscard0m** [reply]
It would be interesting to notify somehow to the user when using a deprecated method. Do you think it woulc be a good idea @gr2m (so I create an issue about it)?

**Reply by gr2m** [reply]
`createBasicAuth` already logs a deprecation message: https://github.com/octokit/auth-basic.js/blob/c4b26ae22c55e5e4afe24590ec0a0d4f5137a26a/src/index.ts#L23-L25

**Reply by oscard0m** [reply]
> `createBasicAuth` already logs a deprecation message: [octokit/auth-basic.js@`c4b26ae`/src/index.ts#L23-L25](https://github.com/octokit/auth-basic.js/blob/c4b26ae22c55e5e4afe24590ec0a0d4f5137a26a/src/index.ts#L23-L25)

Oh! Sorry for not checking it myself. Then we are fine :)
